connect/api/src/main/java/org/apache/kafka/connect/data/SchemaBuilder.java

```diff
@@ -390,6 +390,34 @@ public Schema valueSchema() {
         return valueSchema;
     }
 
+    /**
+     * @param schema the schema to clone the SchemaBuilder from.
+     * @return a new SchemaBuilder from the supplied schema.
+     */
+    public static SchemaBuilder from(Schema schema) {
+        if (null == schema)
+            throw new SchemaBuilderException("schema cannot be null.");
+
+        SchemaBuilder builder = new SchemaBuilder(schema.type());
+        builder.name = schema.name();
+        builder.doc = schema.doc();
+        builder.optional = schema.isOptional();
+        builder.parameters = schema.parameters();
```

Comment:
Some fields might be safe to just shallow copy, but I think a number of these need to be deep copied to avoid accidentally modifying the original schema. I think `parameters and `defaultValue` at a minimum need to change. Schemas seem like they should be fine since they'd just be fully replaced, not modified, anyway.